[15.0][ADD] project_task_note: Add notes in project task

carolinafernandez-tecnativa opened the pull request:
TT47764

@Tecnativa

@pedrobaeza @pilarvargas-tecnativa

carolinafernandez-tecnativa committed:
[15.0][ADD] project_task_note: Add notes in project task
TT47764

pedrobaeza commented on project_task_note/__manifest__.py:
```diff
@@ -0,0 +1,16 @@
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    "name": "Add notes in Project Task",
```
This text is shorter from the summary (or just put the name, not both).

```suggestion
    "name": "Project task notes",
```

pedrobaeza commented on project_task_note/__manifest__.py:
```diff
@@ -0,0 +1,16 @@
+    "name": "Add notes in Project Task",
+    "version": "15.0.1.0.0",
+    "category": "Project Management",
+    "summary": "Add notes in Project Task",
```
```suggestion
    "summary": "Add notes in project tasks",
```

pedrobaeza commented on project_task_note/models/project_task.py:
```diff
@@ -0,0 +1,10 @@
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    notes = fields.Html()
```
```suggestion
    notes = fields.Html(string="Internal notes")
```

pedrobaeza commented on project_task_note/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1 @@
+This module add notes in project task only visible for internal users
```
```suggestion
This module add notes in project tasks, only visible for internal users.
```

Add also a USAGE section for indicating where to see that notes, and maybe a CONTEXT file for indicating why we are creating this (because there's no field where storing general comments and notes not visible to the customers in the portal).

pedrobaeza commented on project_task_note/views/project_task_views.xml:
```diff
@@ -0,0 +1,20 @@
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_form2" />
+        <field name="arch" type="xml">
+            <page name="description_page" position="after">
```
I'm thinking that is better to put the notes in the same tab of the description, as being in read mode, you don't have to switch between pages to see both texts, and they are shortened to the minimum occupied height. Anyway, check how this will see in v16 (always in edit mode).

pilarvargas-tecnativa reviewed (approved):
LGTM :)

pedrobaeza commented:
The way the field widget is hooked seems to be different:

![image](https://github.com/OCA/project/assets/7165771/0e6ca24f-8614-4134-bccd-133e1894da50)


Please use the same as the other field. Have you tested how it looks on v16?

carolinafernandez-tecnativa commented:
> The way the field widget is hooked seems to be different:
> 
> ![image](https://private-user-images.githubusercontent.com/7165771/305114865-0e6ca24f-8614-4134-bccd-133e1894da50.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMTI1MjMsIm5iZiI6MTcwODAxMjIyMywicGF0aCI6Ii83MTY1NzcxLzMwNTExNDg2NS0wZTZjYTI0Zi04NjE0LTQxMzQtYmNjZC0xMzNlMTg5NGRhNTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTVUMTU1MDIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzI3MTIwZWNiNDZiYzU5ODNjOTE5NzNjZmU0YmQ4N2MzZWExZjBiZTkwNWM5NzVmOWIwYmM2M2E5ZjhhOWU2OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.14HfhgzC29rFDf83nHHUvCj79vxsFZXOCobg-uP9gPs)
> 
> Please use the same as the other field. Have you tested how it looks on v16?

I added the same as the other field

https://github.com/odoo/odoo/blob/15.0/addons/project/views/project_views.xml#L898

but stills appear as in your picture, also without group.

On v16 works fine, instead of using group, I added a label because otherwise the field appears as readonly I do not know why..

![image](https://github.com/OCA/project/assets/143796894/2d12f61d-2004-4a96-804f-1f3aa0f7edae)

pedrobaeza commented on project_task_note/views/project_task_views.xml:
```diff
@@ -0,0 +1,20 @@
+        <field name="arch" type="xml">
+            <field name="description" position="after">
+                <group name="internal_notes_group" string="Internal notes">
+                    <field name="notes" type="html" nolabel="1" />
```
You can put the collaborative flag also here.

pedrobaeza reviewed (approved):
/ocabot merge nobump

Migrate it to v16, please
